File: 1833-maximum-ice-cream-bars/1833-maximum-ice-cream-bars.go
func maxIceCream(costs []int, coins int) int {
    iceCream := 0
    totalCost := 0
    sort.Ints(costs)
    for i := 0; i < len(costs); i++ {
        if totalCost + costs[i] > coins {
            return iceCream
        }
        totalCost += costs[i]
        iceCream++
    }
    return iceCream
}

// array, greedy
// time: O(nlog(n))
// space: O(1)

File: 1833-maximum-ice-cream-bars/1833-maximum-ice-cream-bars.py
class Solution:
    def maxIceCream(self, costs: List[int], coins: int) -> int:
        costs.sort()

        ice_cream, total_cost = 0, 0
        for i in range(len(costs)):
            if total_cost + costs[i] > coins:
                return ice_cream
            total_cost += costs[i]
            ice_cream += 1

        return ice_cream

# array, greedy
# time: O(nlog(n))
# space: O(1)